src/esp/nav/PathFinder.cpp

```diff
@@ -1371,11 +1599,13 @@ PathFinder::Impl::getTopDownView(const float metersPerPixel,
   return topdownMap;
 }
 
-assets::MeshData::ptr PathFinder::Impl::getNavMeshData() {
-  if (meshData_ == nullptr && isLoaded()) {
-    meshData_ = assets::MeshData::create();
-    std::vector<esp::vec3f>& vbo = meshData_->vbo;
-    std::vector<uint32_t>& ibo = meshData_->ibo;
+assets::MeshData::ptr PathFinder::Impl::getNavMeshData(
+    int islandIndex /*= ID_UNDEFINED*/) {
+  islandSystem_->assertValidIsland(islandIndex);
+  if (islandMeshData_[islandIndex] == nullptr && isLoaded()) {
```

Comment:
This will actually initialize with the default constructrible value if it doesn't exist (which I believe is nullptr?) You may want to do a proper key check here then and do another emplace.

As an added bonus, a reference to the key:value std::pair created by emplace will be returned so you can just return that instead of looking up the key in the unordered_map again for the return value.